### PR TITLE
test: Don't skip encapsulation tests on GKE

### PR DESCRIPTION
**Pull request by pchaigno**
We can run with VXLAN encapsulation enabled on GKE; there's no need to skip those tests.

### CODE REVIEW

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
test-gke

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -209,10 +205,17 @@ var _ = Describe("K8sDatapathConfig", func() {
...
 				"kubeProxyReplacement": "disabled",
 				"encryption.enabled":   "true",
-			}, DeployCiliumOptionsAndDNS)
+				"tunnel":               "vxlan",
+			}
+			if helpers.RunsOnGKE() {
+				// We need to disable gke.enabled as it disables tunneling.
+				options["gke.enabled"] = "false"
+				options["endpointRoutes.enabled"] = "true"
+			}
```
Comment (tklauser) [inline test/k8sT/DatapathConfiguration.go]: It looks like this bit is repeated across several tests. Would it make sense to move it to a helper?

**Comment by pchaigno** [comment]
test-gke

**Review by tklauser — APPROVED** [review]
Thanks

**Pull request merged**